### ISSUE TITLE
adaptive softmax in transformer-xl

### ISSUE THREAD

**Issue by lepus2** [huggingface/transformers#848]
I guess there is some incomplete part for adaptive softmax in [modeling_transfo_xl.py](https://github.com/huggingface/pytorch-transformers/blob/master/pytorch_transformers/modeling_transfo_xl.py)
Actually, it is impossible to build model not using adaptive softmax even though `TransfoXLConfig` has `adaptive` parameter.
I can see that if `sample_softmax` is larger than -1, model uses sample softmax other than adaptive softmax, which seems to be the case of not using adaptive softmax.
However, in case of `sample_softmax` > -1 and `tie_weight`=True, there is a problem on [this line](https://github.com/huggingface/pytorch-transformers/blob/master/pytorch_transformers/modeling_transfo_xl.py#L1317)<code><pre>self.out_layer.weight = self.transformer.word_emb.weight</pre></code> because the model always use `AdaptiveEmbedding` as `word_emb`, which has no `weight` property.

Presumably we need some code for not using adaptive softmax and but using standard softmax and usual `nn.Embedding` as a word embedding.
Then we can tie weights between standard `nn.Embedding` and `nn.Linear` when we doesn't need adaptive softmax.

Can you consider about this problem? Thanks.

**Comment by thomwolf** [comment]
Yes. At the moment, this library is designed for loading pretrained models mostly and no one has open-sourced a Transformer-XL pretrained model using something else than adaptive softmax so I have not spent time adding these options.

Happy to welcome PR though.

The main thing of interest here, if you want to give it a try would be to add bias to all clusters in PyTorch official [`AdaptiveLogSoftmaxWithLoss`](https://pytorch.org/docs/stable/nn.html?highlight=adaptivelogsoftmaxwithloss#torch.nn.AdaptiveLogSoftmaxWithLoss) module so we could just use the official implementation without maintaining ours.
